client/core/bond.go

```diff
@@ -408,23 +622,68 @@ func deriveBondKey(seed []byte, assetID, bondIndex uint32) (*secp256k1.PrivateKe
 	return priv, nil
 }
 
-func (c *Core) nextBondKey(crypter encrypt.Crypter, assetID uint32) (*secp256k1.PrivateKey, error) {
-	creds := c.creds()
-	if creds == nil {
-		return nil, errors.New("app not initialized")
-	}
-	seed, err := crypter.Decrypt(creds.EncSeed)
-	if err != nil {
-		return nil, fmt.Errorf("seed decryption error: %w", err)
-	}
-	defer encode.ClearBytes(seed)
+func deriveBondXPriv(seed []byte) (*hdkeychain.ExtendedKey, error) {
+	return keygen.GenDeepChild(seed, []uint32{hdKeyPurposeBonds})
+}
 
+func (c *Core) nextBondKey(assetID uint32) (*secp256k1.PrivateKey, error) {
 	nextBondKeyIndex, err := c.db.NextBondKeyIndex(assetID)
 	if err != nil {
 		return nil, fmt.Errorf("NextBondIndex: %v", err)
 	}
 
-	return deriveBondKey(seed, assetID, nextBondKeyIndex)
+	c.loginMtx.Lock()
+	defer c.loginMtx.Unlock()
+
+	if c.bondXPriv == nil {
+		return nil, errors.New("not logged in")
+	}
+
+	return deriveBondKey(c.bondXPriv, assetID, nextBondKeyIndex)
+}
+
+// UpdateBondOptions sets the bond rotation options for a DEX host, including
+// the target trading tier, the preferred asset to use for bonds, and the
+// maximum amount allowable to be locked in bonds.
+func (c *Core) UpdateBondOptions(form *BondOptionsForm) error {
+	dc, _, err := c.dex(form.Addr)
+	if err != nil {
+		return err
+	}
+	acct, err := c.db.Account(form.Addr)
+	if err != nil {
+		return err
+	}
+	dc.acct.authMtx.Lock()
+	defer dc.acct.authMtx.Unlock()
+	// Verify the new bond asset wallet first.
+	if form.BondAsset != nil {
+		bondAssetID := *form.BondAsset
+		wallet, found := c.wallet(bondAssetID)
+		if !found || !wallet.connected() {
+			return fmt.Errorf("bond asset wallet %v does not exist or is not connected", unbip(bondAssetID))
+		}
+		if _, ok := wallet.Wallet.(asset.Bonder); !ok {
+			return fmt.Errorf("wallet %v is not an asset.Bonder", unbip(bondAssetID))
+		}
+		_, err = wallet.refreshUnlock()
+		if err != nil {
+			return fmt.Errorf("bond asset wallet %v is locked", unbip(bondAssetID))
+		}
+		dc.acct.bondAsset = bondAssetID
+		acct.BondAsset = bondAssetID
+	}
+	if form.TargetTier != nil {
+		dc.acct.targetTier = *form.TargetTier
+		acct.TargetTier = *form.TargetTier
+	}
+	if form.MaxBondedAmt != nil {
+		dc.acct.maxBondedAmt = *form.MaxBondedAmt
+		acct.MaxBondedAmt = *form.MaxBondedAmt
+	}
+	c.log.Debugf("Bond options for %v: target tier %d, bond asset %d, maxBonded %v",
+		acct.Host, dc.acct.targetTier, dc.acct.bondAsset, acct.MaxBondedAmt)
+	return c.db.UpdateAccountInfo(acct)
```

Comment:
If updating the db fails, should dc.acct revert to old values? Or not even set the new values until db update succeeds.